chingu-voyage4/Geckos-Team-32
# Link backend to frontend for user data

## Purpose
* Connect backend with frontend

## Approach/Fixes
* Merge code from `development` branch, `pages/auth` branch, and `database` branch
* Use `withRouter` for a page redirect
* Use axios GET request to obtain user login info from backend server-side code
* Save user info in state in `AppRouter.jsx` file
* Present user data in dashboard component IF logged in, otherwise show static info
* Re-factored dashboard component to split up files for modularity
* In the `package.json` file I added a new script that utilizes `nodemon`
  * You don't have to run this script, but if you want to just run `npm run serve` to have nodemon watch your server-side changes and have webpack watched automatically

## Testing
* Download this PR to your local device.
* Run `mongod` in your terminal to connect to your local mongoDB
* Run `npm start` to run the app on our server (localhost:8000)
* If you haven't yet already, create a new user
  * If you create a new user, I haven't configured the code to handle that as a proper login yet
  * Refresh, and login using the newly created user
* If you have already created a user, login using your login credentials
* Notice that there is a 'middle' page that re-directs to the homepage as it gathers your user info
* The left-hand side dashboard should not be updated to show your username
  * When we add more to the user schema it'll show more info

## Notes
* I had to create a 'middle' routing page to get the data from the back end, then redirect to the homepage in order to properly access the url params for the user ID
* @marylicious this PR doesn't include your new code from the `selectVideo` branch so feel free to add that to this afterwards if you've finalized things!
* We will still need a logout button to handle logging out
* I had to add this code to the html file: `<base href="/" />` to properly handle the routing... NOT SURE exactly what it's doing but I THINK it's just adding a '/' to the url route to that it's handled correctly...
